src/Processors/QueryPlan/Optimizations/optimizeJoin.cpp

```diff
@@ -505,12 +505,6 @@ size_t addChildQueryGraph(QueryGraphBuilder & graph, QueryPlan::Node * node, Que
 
     graph.inputs.push_back(node);
     RelationStats stats = estimateReadRowsCount(*node);
```

Comment:
The issue is that estimateReadRowsCount already returns dummy stats from hint for Read step: https://github.com/ClickHouse/ClickHouse/blob/eaf7fd0bb8382854381e9e3fc6f694a99776a8ae/src/Processors/QueryPlan/Optimizations/optimizeJoin.cpp#L205-L206

An Expression step on top of read remap unqualified column names like `p_partkey` to qualified names like `__table2.p_partkey` that are used in the Join step above:
https://github.com/ClickHouse/ClickHouse/blob/eaf7fd0bb8382854381e9e3fc6f694a99776a8ae/src/Processors/QueryPlan/Optimizations/optimizeJoin.cpp#L264-L267

These deleted lines were overwriting correct qualified name back to unqualifeid